### PR TITLE
Update AddHoneycomb with settings instance to use Action<>

### LINKED ISSUE #17: X-Honeycomb-Team not set
When giving an instance of HoneycombApiSettings in a ASP.NET core application, the X-Honeycomb-Team is not set in the HTTP header (inspected using Fiddler).
```
services.AddHoneycomb(new HoneycombApiSettings {
						TeamId = "xyz",
						DefaultDataSet = "abc",
						BatchSize = 1,
						SendFrequency = 5000,
				});
```
However, putting the settings in appsettings works:
```
services.AddHoneycomb(Configuration);
```

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Fixes #17.

Updates the `serviceCollection.AddHoneycomb` programatic overload to take an `Action<HoneycombApiSettings>`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,25 +64,26 @@ Configuration can either be done through adding this to your appSettings.json
...
     ...
 
-    services.AddHoneycomb(new HoneycombApiSettings {
-        TeamId = "blah",
-        DefaultDataSet = "MyTestDataSet"
-        BatchSize = 100,
-        SendFrequency = 10000,
+    services.AddHoneycomb(options => {
+        options.ApiHost = "https://api.honeycomb.io";
+        options.TeamId = "<your-writekey>";
```
Comment (vreynolds) [inline README.md:83]: Based on the other merged PR, this should now be WriteKey

**Comment by ericsampson** [comment]
Out of curiosity, is there a precedence/convention of not supporting both options (class and Action) ? I've seen different libraries choose different combinations of `IConfiguration`, `LibSettings`, and `Action<LibSettings>configureLibSettings`

**Comment by MikeGoldsmith** [comment]
> Out of curiosity, is there a precedence/convention of not supporting both options (class and Action) ? I've seen different libraries choose different combinations of `IConfiguration`, `LibSettings`, and `Action<LibSettings>configureLibSettings`

There isn't a convention of preferring one type of parameter vs another. I personally find the `Action<T>` style the better solution because the library is in control of creating the settings instance which i handed to the function. I've used both a class instance and the `Action<>` together in the past, where the `Action<T>` just creates the settings instance and then passes into the other method.

**Review by vreynolds — APPROVED** [review]

**Comment by ericsampson** [comment]
@MikeGoldsmith what I noticing is that the [way it was originally](https://github.com/honeycombio/libhoney-dotnet/pull/29/files#diff-44ba2272f3fb3eb2c14f0a69e2ee2443a1a555a4469ee3984d497fe7eaca9a18L21) doesn't work, unless I'm being extra slow today?

Here's an example demonstrating how it was before, and what happens.
image

**Comment by MikeGoldsmith** [comment]
> @MikeGoldsmith what I noticing is that the [way it was originally](https://github.com/honeycombio/libhoney-dotnet/pull/29/files#diff-44ba2272f3fb3eb2c14f0a69e2ee2443a1a555a4469ee3984d497fe7eaca9a18L21) doesn't work, unless I'm being extra slow today?
> 
> Here's an example demonstrating how it was before, and what happens.
> image

That's correct, the previous way of calling `services.Configure(settings);` didn't work. The updated version passing an `Action<HoneycombApiSettings>` to configure does work. I believe it's to do with scoping of the services DI graph, but not 100% certain. I've used it previously with an Action<T> and it worked here too 🤷‍♂️ .

**Pull request merged**